sycl/include/CL/sycl/accessor.hpp

```diff
@@ -325,6 +325,14 @@ class accessor_common {
       return MAccessor[MIDs];
     }
 
+    template <int CurDims = SubDims>
+    typename detail::enable_if_t<
+        CurDims == 1 && AccessMode == access::mode::atomic, atomic<DataT, AS>>
+    operator[](size_t Index) const {
+      MIDs[Dims - CurDims] = Index;
+      return MAccessor[MIDs];
```

Comment:
should it be some explicit construction of an object of `atomic<DataT, AS>` type?

Reply:
The sycl::atomic is constructed in the accessor class, when we call MAccessor[MIDs]. We do not create any objects in AccessorSubscript::operator[] methods, only call operator[] for the accessor. I think it is more correct to create sycl::atomic out of this class.

Reply:
Sounds good, thanks